ir/pattern/constraint.rs

```diff
@@ -162,88 +164,88 @@ impl Display for Constraints {
 }
 
 #[derive(Debug, Clone, PartialEq)]
-pub enum Constraint {
-    Type(Type),
-    Isa(Isa),
-    RolePlayer(RolePlayer),
-    Has(Has),
-    ExpressionBinding(ExpressionBinding),
-    FunctionCallBinding(FunctionCallBinding),
-    Comparison(Comparison),
-}
-
-impl Constraint {
-    pub fn variables(&self) -> Box<dyn Iterator<Item = Variable> + '_> {
+pub enum Constraint<ID: IrID> {
```

Comment:
IR constraints are parametrised, so we can swap out the from/to in the execution layer to avoid extra indirection during traversal.